### PR TITLE
Add file-open command to file nodes

### PR DESCRIPTION
**Pull request by nickcabral**
This change adds "jump to file" functionality when file nodes in the tag tree are clicked. The file is opened and focused if it isn't yet open in a tab, or just focused if it is already open in a tab.

I also made the following changes:
- Make treeDataProvider use vscode the findFiles function for searching.
- Read files and scan for tags asynchronously.

I haven't yet added tests around the change(s). There don't seem to be existing cases around the class I touched, so maybe that's fine for now.

### CODE REVIEW

**Comment by nickcabral** [comment]
After re-reading #3, I'm thinking the changes here don't implement that request. This change just opens the file when a tag is clicked in the tree, but it doesn't specifically "go to each of the occurrences of a tag in the file". I can add that as part of this PR or as follow-up, whichever you prefer. The functionality I have here does seem useful independent of the preferred "go to a specific line in the file".

**Comment by blakedietz** [comment]
You're right. However, I would rather keep the pr as is and we can do another pr to get the outcome requested in #3.

File: src/tag-tree-data-provider.ts [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import { debounce } from "debounce";
 import * as fs from "fs";
-import * as recursiveReadDir from "recursive-readdir";
```
Comment (blakedietz) [inline src/tag-tree-data-provider.ts]: We'll need to pull this out of package.json but I'm not going to block your pr because of that.

Comment (nickcabral) [reply src/tag-tree-data-provider.ts]: Good call, I missed that.

**Pull request merged**